test/rubygems/test_gem_gem_runner.rb

```diff
@@ -11,12 +11,25 @@ def setup
 
     require 'rubygems/gem_runner'
     @runner = Gem::GemRunner.new
+
+    singleton_gem_class.class_eval do
+      alias_method :orig_latest_rubygems_version, :latest_rubygems_version
+
+      def latest_rubygems_version
+        Gem.rubygems_version
+      end
+    end
   end
 
   def teardown
     super
 
     Gem::Command.build_args = @orig_args
+
+    singleton_gem_class.class_eval do
```

Comment:
Maybe we can include this logic in `test_warn_on_outdated` since we just need it for that test

Reply:
I gave that a go but doing it caused tests to fail
```
TestGemGemRunner#test_query_is_deprecated:
RuntimeError: Can't find 'rubygems-update' in any repo. Check `gem source list`.
    /__w/rubygems/rubygems/lib/rubygems.rb:909:in `latest_rubygems_version'
    /__w/rubygems/rubygems/lib/rubygems/gem_runner.rb:86:in `warn_on_outdated'
    /__w/rubygems/rubygems/lib/rubygems/gem_runner.rb:60:in `run'
    /__w/rubygems/rubygems/test/rubygems/test_gem_gem_runner.rb:74:in `block in test_query_is_deprecated'
    /__w/rubygems/rubygems/lib/rubygems/user_interaction.rb:47:in `use_ui'
    /__w/rubygems/rubygems/lib/rubygems/user_interaction.rb:70:in `use_ui'
    /__w/rubygems/rubygems/test/rubygems/test_gem_gem_runner.rb:73:in `test_query_is_deprecated'

```

It seems like the gem `rubygems-update` needs to be around in order to successfully determine `Gem.latest_rubygems_version`. eg.
https://github.com/rubygems/rubygems/blob/master/lib/rubygems.rb#L907-L910 

Adding this to the setup/teardown of the tests get around this problem. Maybe there is a nicer way to solve this though? Could also have a try/catch type thing so that it does not fail tragically when `rubygems-update is not available`?